src/main/java/lotto/service/LottoServiceImpl.java

```diff
@@ -0,0 +1,110 @@
+                    .map(Integer::parseInt)
+                    .collect(Collectors.toList());
+
+        } catch (NumberFormatException error) {
+            throw new IllegalArgumentException(NUMBER_FORMAT_ERROR_MESSAGE);
+        }
+    }
+
+    public List<Rank> getLottoWinnings(List<Lotto> purchaseLottos, Lotto winningLotto, LottoNumber bonusNumber) {
+        List<Rank> winnings = new ArrayList<>();
+        List<LottoNumber> winningNumbers = winningLotto.getNumbers();
+
+        for (Lotto purchaseLotto : purchaseLottos) {
+            Optional<Rank> lottoWinning = getLottoWinning(bonusNumber, winningNumbers, purchaseLotto);
+            if (lottoWinning.isEmpty()) {
+                continue;
+            }
+            winnings.add(lottoWinning.get());
+        }
+
+        return winnings;
+    }
+
+    public Map<Rank, Integer> getWinnings(List<Rank> ranks) {
+        Map<Rank, Integer> winnings = createWinningMap();
+        applyWinningData(ranks, winnings);
+
+        return winnings;
+    }
+
+    public float getProfitPercentage(Map<Rank, Integer> winnings, Money money) {
+        return calculateProfitPercentage(money, calculateProfit(winnings));
+    }
+
+
+    private Optional<Rank> getLottoWinning(LottoNumber bonusNumber, List<LottoNumber> winningNumbers, Lotto purchaseLotto) {
+        int matchCount;
+        List<LottoNumber> purchaseLottoNumbers = purchaseLotto.getNumbers();
+
+        matchCount = getWinningCount(winningNumbers, purchaseLottoNumbers);
+        if (matchCount < FIFTH_PRIZE_COUNT) {
+            return Optional.empty();
+        }
+        return Optional.of(getLottoWinning(matchCount, purchaseLottoNumbers.contains(bonusNumber)));
+    }
+
+    private int getWinningCount(List<LottoNumber> winningNumbers, List<LottoNumber> purchaseLottoNumbers) {
+        int winningCount = 0;
+
+        for (LottoNumber lottoNumber : purchaseLottoNumbers) {
+            if (winningNumbers.contains(lottoNumber)) {
+                winningCount += 1;
+            }
+        }
+        return winningCount;
+    }
+
+    private Rank getLottoWinning(int matchCount, boolean isBonusNumberMatched) {
+        return Rank.findLottoWinning(matchCount, isBonusNumberMatched);
+    }
+
+
+    private void applyWinningData(List<Rank> ranks, Map<Rank, Integer> winnings) {
+        for (Rank rank : ranks) {
+            winnings.replace(rank, winnings.get(rank)+1);
+        }
+    }
+
+    private Map<Rank, Integer> createWinningMap() {
+        Map<Rank, Integer> winnings = new LinkedHashMap<>();
+        for (Rank rank : Rank.values()) {
+            winnings.put(rank, 0);
+        }
+        return winnings;
+    }
+
+    private float calculateProfit(Map<Rank, Integer> winnings) {
+        float profit = 0.0f;
+
+        for (Rank rank : Rank.values()) {
+            Integer winningCount = winnings.get(rank);
+            profit += rank.getReward() * winningCount;
```

Comment:
getter를 사용하기보다 rank에서 작업하도록 하는 것은 어떠신가요?
```suggestion
            profit += rank.calculateProfit();
```

ranks 라는 일급 컬렉션을 두면 관리하기 더 편할 것 같아요:)

Reply:
> getter를 사용하기보다 rank에서 작업하도록 하는 것은 어떠신가요?

정말 좋은 의견입니다!!!!! 감사합니다!

Reply:
> ranks 라는 일급 컬렉션을 두면 관리하기 더 편할 것 같아요:)
일급 컬렉션이 필요한 부분의 꽤 있네요!! 한번 적용해보도록 하겠습니다 감사합니다!